src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityFurnace.java

```diff
@@ -162,22 +169,35 @@ public float getTempRate() {
 
     @Override
     public void update() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "update" is 33 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)